# Aviation Accidents Analysis

## 1. Overview

A company I am working with is planning to diversify its portfolio by exploring new areas, particularly the aviation industry. The leadership is interested in the risks associated with aircraft operations, mainly safety and reliability. It is on that background that as a senior data scientist in the company that the management tusked me with the responsibility of carrying out data analysis to assist the management in making informed investment decisions to determine which aircraft presents the lowest operational risk.


## 2. Business Understanding
To find out the safest and most reliable aircraft for the company to purchase based on an analysis of accident and incident data collected over the years. Therefore, the company wants to:
* Minimize accident frequency
* low fetality count


## 3. Data Science Understanding
Analyze accident data collected over the years to establish which aircraft has the lowest risk by using risk indicators such as accident frequency, fatality rate, and severity of injuries.
This will involve the following analysis:
* Data exploration
* Cleaning
* Grouping aircraft by make and model
* Calculating the safety matrix per aircraft type
* Visualizing patterns of high-risk and low-risk aircrafts


In [3]:
# importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [7]:
# Loading accident data to the notebook
aviation_data = pd.read_csv("../Data/AviationData.csv", index_col = 0, encoding = "latin1", low_memory = False)

### 3.1 exploring data

In [8]:
aviation_data.head() # checking the first five rows of the dataframe

Investigation.Type Accident.Number  Event.Date  \
Event.Id                                                        
20001218X45444           Accident      SEA87LA080  1948-10-24   
20001218X45447           Accident      LAX94LA336  1962-07-19   
20061025X01555           Accident      NYC07LA005  1974-08-30   
20001218X45448           Accident      LAX96LA321  1977-06-19   
20041105X01764           Accident      CHI79FA064  1979-08-02   

                       Location        Country   Latitude   Longitude  \
Event.Id                                                                
20001218X45444  MOOSE CREEK, ID  United States        NaN         NaN   
20001218X45447   BRIDGEPORT, CA  United States        NaN         NaN   
20061025X01555    Saltville, VA  United States  36.922223  -81.878056   
20001218X45448       EUREKA, CA  United States        NaN         NaN   
20041105X01764       Canton, OH  United States        NaN         NaN   

               Airport.Code Airport.Name Injury.Severity  ...  \
Event.Id                                                  ...   
20001218X45444          NaN          NaN        Fatal(2)  ...   
20001218X45447          NaN          NaN        Fatal(4)  ...   
20061025X01555          NaN          NaN        Fatal(3)  ...   
20001218X45448          NaN          NaN        Fatal(2)  ...   
20041105X01764          NaN          NaN        Fatal(1)  ...   

               Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
Event.Id                                                            
20001218X45444          Personal         NaN                  2.0   
20001218X45447          Personal         NaN                  4.0   
20061025X01555          Personal         NaN                  3.0   
20001218X45448          Personal         NaN                  2.0   
20041105X01764          Personal         NaN                  1.0   

               Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
Event.Id                                                                     
20001218X45444                    0.0                  0.0             0.0   
20001218X45447                    0.0                  0.0             0.0   
20061025X01555                    NaN                  NaN             NaN   
20001218X45448                    0.0                  0.0             0.0   
20041105X01764                    2.0                  NaN             0.0   

                Weather.Condition Broad.phase.of.flight   Report.Status  \
Event.Id                                                                  
20001218X45444                UNK                Cruise  Probable Cause   
20001218X45447                UNK               Unknown  Probable Cause   
20061025X01555                IMC                Cruise  Probable Cause   
20001218X45448                IMC                Cruise  Probable Cause   
20041105X01764                VMC              Approach  Probable Cause   

               Publication.Date  
Event.Id                         
20001218X45444              NaN  
20001218X45447       19-09-1996  
20061025X01555       26-02-2007  
20001218X45448       12-09-2000  
20041105X01764       16-04-1980  

[5 rows x 30 columns]

In [9]:
aviation_data.tail() # chwcking the last five rows of the dataframe

Investigation.Type Accident.Number  Event.Date       Location  \
Event.Id                                                                       
20221227106491           Accident      ERA23LA093  2022-12-26  Annapolis, MD   
20221227106494           Accident      ERA23LA095  2022-12-26    Hampton, NH   
20221227106497           Accident      WPR23LA075  2022-12-26     Payson, AZ   
20221227106498           Accident      WPR23LA076  2022-12-26     Morgan, UT   
20221230106513           Accident      ERA23LA097  2022-12-29     Athens, GA   

                      Country Latitude Longitude Airport.Code Airport.Name  \
Event.Id                                                                     
20221227106491  United States      NaN       NaN          NaN          NaN   
20221227106494  United States      NaN       NaN          NaN          NaN   
20221227106497  United States  341525N  1112021W          PAN       PAYSON   
20221227106498  United States      NaN       NaN          NaN          NaN   
20221230106513  United States      NaN       NaN          NaN          NaN   

               Injury.Severity  ... Purpose.of.flight         Air.carrier  \
Event.Id                        ...                                         
20221227106491           Minor  ...          Personal                 NaN   
20221227106494             NaN  ...               NaN                 NaN   
20221227106497       Non-Fatal  ...          Personal                 NaN   
20221227106498             NaN  ...          Personal  MC CESSNA 210N LLC   
20221230106513           Minor  ...          Personal                 NaN   

               Total.Fatal.Injuries Total.Serious.Injuries  \
Event.Id                                                     
20221227106491                  0.0                    1.0   
20221227106494                  0.0                    0.0   
20221227106497                  0.0                    0.0   
20221227106498                  0.0                    0.0   
20221230106513                  0.0                    1.0   

               Total.Minor.Injuries Total.Uninjured  Weather.Condition  \
Event.Id                                                                 
20221227106491                  0.0             0.0                NaN   
20221227106494                  0.0             0.0                NaN   
20221227106497                  0.0             1.0                VMC   
20221227106498                  0.0             0.0                NaN   
20221230106513                  0.0             1.0                NaN   

               Broad.phase.of.flight Report.Status Publication.Date  
Event.Id                                                             
20221227106491                   NaN           NaN       29-12-2022  
20221227106494                   NaN           NaN              NaN  
20221227106497                   NaN           NaN       27-12-2022  
20221227106498                   NaN           NaN              NaN  
20221230106513                   NaN           NaN       30-12-2022  

[5 rows x 30 columns]

In [12]:
aviation_data.info() # getting concise information about the dataframe

<class 'pandas.core.frame.DataFrame'>
Index: 88889 entries, 20001218X45444 to 20221230106513
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Investigation.Type      88889 non-null  object 
 1   Accident.Number         88889 non-null  object 
 2   Event.Date              88889 non-null  object 
 3   Location                88837 non-null  object 
 4   Country                 88663 non-null  object 
 5   Latitude                34382 non-null  object 
 6   Longitude               34373 non-null  object 
 7   Airport.Code            50132 non-null  object 
 8   Airport.Name            52704 non-null  object 
 9   Injury.Severity         87889 non-null  object 
 10  Aircraft.damage         85695 non-null  object 
 11  Aircraft.Category       32287 non-null  object 
 12  Registration.Number     87507 non-null  object 
 13  Make                    88826 non-null  object 
 14  Model                

All the columns have missing values except the first three.

In [11]:
aviation_data.shape # checking the dataframe dimensions, i.e number of rows and cloumns in the dataframe.

(88889, 30)

Dataset contains 88,889 rows and 30 columns.

## 4. Data Preparation

## 5. Exploratory Data Analysis

## 6. Recommendations